src/aind_ephys_rig_qc/qc_figures.py

```diff
@@ -54,18 +70,164 @@ def plot_power_spectrum(data, sample_rate, stream_name):
     matplotlib.figure.Figure
         Figure object containing the plot
     """
-
     fig = Figure(figsize=(10, 4))
-    ax = fig.subplots()
+    stream_names, _ = se.get_neo_streams("openephys", directory)
+    target_stream = [
+        curr_stream_name
+        for curr_stream_name in stream_names
+        if stream_name in curr_stream_name
+    ][0]
+
+    recording = se.read_openephys(
+        folder_path=directory, stream_name=target_stream,
+    )
+
+    ax = fig.subplots(
+        2, num_chunks, gridspec_kw={"height_ratios": [2, 1]}, sharex=True
+    )
+    sample_rate = recording.get_sampling_frequency()
+    subsets = si.get_random_data_chunks(
+        recording, num_chunks_per_segment=num_chunks, chunk_size=chunk_size
+    )
+    for chunk_ind in range(num_chunks):
+        subset = subsets[
+            (chunk_ind * chunk_size): ((chunk_ind + 1) * chunk_size), :
+        ]
```

Comment:
@ZhixiaoSu is you want multiple chunks of `chunk_size`, you can add the `concatenated=False` param to the `si.get_random_data_chunks`

Reply:
Thank you! I have been looking around how to do it.